src/main/java/frc/robot/AutoRotateUtil.java

```diff
@@ -0,0 +1,63 @@
+package frc.robot;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.subsystems.Swerve;
+
+public class AutoRotateUtil {
+    
+    private final Swerve s_Swerve;
+    private final PIDController pidController;
+
+    private int m_angle;
+    public AutoRotateUtil(Swerve s_Swerve, int angle) {
+        this.s_Swerve = s_Swerve;
+        this.m_angle = angle == 0?360:angle;
+
+        //SmartDashboard.putNumber("Angle", this.m_angle);
+        this.pidController = new PIDController(0, 0, 0);
+
+        pidController.setTolerance(0.675);
+        pidController.setSetpoint(this.m_angle);
+        SmartDashboard.putNumber("kP", 0.0065);
+        SmartDashboard.putNumber("kI", 0.0025);
+        SmartDashboard.putNumber("kD", 0.001);
+   }
+
+   public void initialize() {
+    pidController.reset();
+   }
+
+
+   public double calculateRotationSpeed () {
+    
+    double kP = SmartDashboard.getNumber("kP", 0.0);
```

Comment:
It looks like the formatting is off in this file. You might try running vs code's formatter on it.

Reply:
now formatted here:
https://github.com/CSE-FortOsageCTC/5801_Inspire_2024/pull/26/commits/a0fc0d70d2df62420bfdbb0501ab99ed792dd0cf